projects/angular-calendar/src/modules/month/calendar-month-row.component.ts

```diff
@@ -149,54 +137,52 @@ export class CalendarMonthRowComponent implements OnChanges {
 
   manageWidth(note: CalendarEvent): string {
     let width: number;
-    const indexStartDate = this.daysSliced
-      .map((daySliced) =>
-        this.timezoneManager.reverseTz(daySliced.date, this.timezone)
-      )
-      .findIndex(
-        (date) =>
-          new Date(date.toDateString()).valueOf() === note.start.valueOf()
-      );
-    const indexEndDate = this.daysSliced
-      .map((daySliced) =>
-        this.timezoneManager.reverseTz(daySliced.date, this.timezone)
-      )
-      .findIndex(
-        (date) => new Date(date.toDateString()).valueOf() === note.end.valueOf()
-      );
-    // start et end dans le row en cours
-    if (indexEndDate !== -1 && indexStartDate !== -1) {
-      width = (indexEndDate - indexStartDate + 1) * (100 / 7);
+    const oneDayInMs = 1000 * 60 * 60 * 24;
+    const weekEndDate = new Date(
+      this.daysSliced[6].date.getTime() + oneDayInMs - 1
+    );
```

Comment:
If we need such calculation we should rely on timezoneManager that is timezone compliant, day don't have 24 hours duration.

Reply:
i have used moment to do this and seems to work just fine 
`````
const noteEnd = moment.tz(note.end, this.timezone).add(1, 'd').toDate();
        const weekEnd = moment
          .tz(this.endDate, this.timezone)
          .add(1, 'd')
          .toDate();